### DISCUSSION TITLE
Welcome to TopoToolbox Discussions!

### DISCUSSION THREAD

**Discussion by wschwanghart** [wschwanghart/topotoolbox#28] [Announcements]
## 👋 Welcome TTBoxers!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by AllanLopez** [comment]
I everybody. Allan López from Costa Rica, a small but highly deformed crust among the Cocos, Caribbean and Nazca plates.Here TTB  is helping me a lot to unravell and descipher this morphotectonic mess. This New  link is deeply apprecciated thanks Wolfgang

**Comment by fhz-3722** [comment]
Hello Dr. Wolfgang, 
can I ask you to write a tutorial? I want to represent several rivers in one figure, just like in Figure 2B in the paper (https://doi.org/10.1130/L150.1). In a region where mainly one big river and its tributaries are developing, how can I use Topotoolbox to represent the big tributaries instead of other small tributaries? 
I am sorry to bother you with this very simple question. Or you can just give me some suggestions on how to use it.

All the best.

**Reply by wschwanghart** [reply]
Hi, 
thanks for your question. There are several functions that enable you to (programmatically or manually) modify the stream network so that it contains only the streams that you want to visualize or analyze, e.g. `trunk`, `modify`, `klargestconncomps`.
Once you have prepared the network, you can plot it using `plotdz`. 
Note that it is also possible to assign different colors when using `plotdz` with the option **color**. Map views of colored stream networks can be obtained with the function `plotc`.
Hope this helps.
Wolfgang

**Reply by wschwanghart** [reply]
Also, please start a separate discussion if you ask questions here. It will make it easier for others to find questions and answers.

**Reply by fhz-3722** [reply]
Thanks for your reply. I remember your suggestions and will open a new discussion next time.

All best to your future research.

**Comment by sraul1** [comment]
Hey all, I like the decision to switch over to discussions from the blog!

Wolfgang, you might recognize my name from unsolicited emails and blog questions but for others I am finishing up my PhD in Hydrology from the University of Georgia. I mostly use TTB for stream network analysis and surficial groundwater modeling, with an emphasis on using its abilities to structure surface drainage networks as DAGs in ways I can't _quite_ capture using R or Python packages designed for graph network analysis.

Looking forward to see the conversations that come from this setup!

**Reply by wschwanghart** [reply]
Hey Scott,
great you are here. And yes, any talk/exchange about network analysis (in whatever software) is highly appreciated. 
Cheers, Wolfgang

**Comment by AllanLopez** [comment]
Dear Wolfgang. I need to export the drainage basin asymmetry output map as SHP and when I apply: shapewrite(DB,'NAME') I only get the individual basins without the right and left sides of the trunks. I will deeply appreciate if you can modify the shapewrite instruction or code to get the tilted sides and if possible, to identify them with a shadow or ticker outline.  I have unsuccessfully tried several combinations. Thank you very much in advance. Allan Lopez

**Reply by wschwanghart** [reply]
Hi Allan, better start a new discussion/question so that your question can also be found by others. Currently, the individual sides are not returned as grids or polygon, but here's some code so that this can be done easily.
```
DEM = GRIDobj('srtm_bigtujunga30m_utm11.tif');
FD  = FLOWobj(DEM);
S   = STREAMobj(FD,'minarea',1000);
[D,AS,ST] = dbasymmetry(FD,S);

% Set D (half catchments) to either 0 or 1 (in the output, the stream has
% the value 0.5).
D.Z = uint32(D.Z>0);
% Compute drainage basins
DB  = drainagebasins(FD,ST);
% Multiply drainage basin index and add D. Hence, there is a unique value
% for every half catchment.
DB  = DB*2 + D;

% Compute polygon
MS  = GRIDobj2polygon(DB);
shapewrite(MS,'half_catchments.shp')
```
Hope this works.
Best, Wolfgang

**Reply by AllanLopez** [reply]
Dear Wolfgang, It worked fine and helped me a lot. You are right about this item can benefit others. I was confused where to post it. Many thanks'. Allan

**Comment by xiaoxie66621** [comment]
Dr. Wolfgang：
      Hello! I'm a newbie to use Topotoolbox,and I've been studying Topotoolbox recently. I encountered some difficulties. Could you help me? I want to create a simple terrain（dem）like the one presented in your usersguide ，and this simple terrain could contains elevation values and construction information,but i do not find way to solve it ,  Could you just give me some suggestions on how to solve it？
      I am so sorry to occupy your time to help me solve this very simple question.
All the best

**Reply by wschwanghart** [reply]
Hi,
TopoToolbox is commonly used as a tool to work with existent DEMs rather than constructing them, e.g. by point cloud processing and interpolation. I also have no idea what your input data does look like. Perhaps you might to provide more information about your data and the expected outcome.
Best regards, Wolfgang

**Reply by xiaoxie66621** [reply]
Dr. Wolfgang：
I am very sorry for not being able to explain the issue to you clearly last time. Thank you for your reply. I have already resolved the issue last time and can build the basic terrain in TTLEM. But I have another question. In TTLEM, time steps can be set so that the terrain begins to change. What code can I use to extract DEM data for each time step (for example, if I have 4 million years and the terrain rises every 50000 years, how can I extract DEM data after every 50000 years)? I would greatly appreciate it if you could reply。
All the best

**Reply by wschwanghart** [reply]
Hi, you can save the simulated DEMs to your disk by defining a couple of options using `ttlemset` incl.
```
'saveeach'     {inf}. Saves the modelled DEM as GRIDobj each nth time
                     step.
'fileprefix'   {'res_'}. Includes this prefix before each numbered
                     .mat file when saving the data to the harddrive.
'resultsdir'   output directory {['results' filesep]}
``` 
These files are saved as mat-files which contain the GRIDobj of the DEM.
All the best, Wolfgang

**Reply by xiaoxie66621** [reply]
Thank you very much for answering my question. Your answer has been of great help to me. However, I found a problem during the setup (not very clear). I set 0.5 Ma as the interval for lifting once, and plotted DEM according to each time step. When I plotted the evolution map for 1 Ma, the resulting DEM data size value was 1055 × 1447 is different from other DEMs (others are 666 ×666), What is the reason for this problem?

**Comment by 114514-GGGG** [comment]
Hello Dr. Wolfgang，
  I created a basic terrain using ttlem and also set up differential uplift, but I couldn't find knickpoints in that terrain （I used knickpointfinder）. Could you provide some ideas on how to find it?Thank you very much for answering my question.
 
All best to your future research.

**Reply by wschwanghart** [reply]
Oh, I realized I forgot to answer here. What about the tolerance of knickpointfinder? Maybe it's to high?
